### PR TITLE
Add cudnn.BatchNormalization and cudnn.VolumetricBatchNormalization

### PR DESCRIPTION
**Pull request by colesbury**
See https://github.com/torch/nn/pull/665 and https://github.com/torch/cunn/pull/224

Also see issues #117 and PRs #94 and #119

### CODE REVIEW

**Comment by mys007** [comment]
An elegant solution with code reuse!
